[Improvement-16507] Remove spring-boot-starter-cache

binitshrest opened the pull request:
I changed and removed the some extra dependencies that should be removed after removing spring-boot-starter-cache from pom.xml and application.properties file. That has caused the issues. After removing extra dependencies and the server runs smoothly with out any bugs

close #16507

binitshrest committed:
Removed Spring-boot-starter-cache dependencies

binitshrest committed:
Removed Spring-boot-starter-cache dependencies

binitshrest committed:
Merge remote-tracking branch 'origin/dev' into dev

binitshrest committed:
Merge remote-tracking branch 'origin/dev' into dev

binitshrest committed:
Merge branch 'dev' into dev

binitshrest commented:
> Thanks for opening this pull request! Please check out our contributing guidelines. (https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)
Hi, Can  you please review if i updated based on the given documentations.

ruanwenjun reviewed (commented):
We should remove the code and dependency of spring-boot-starter-cache.

binitshrest commented:
> We should remove the code and dependency of spring-boot-starter-cache.

Hi ruanwenjun,
After carefully observing and understanding  the codebase. I have removed the extra dependency of the spring-boot-starter-cache which includes the related annotations like cacheable and cacheconfig. I removed it and the codebase runs without any issues.  I am open for some more suggestions from your side. As spring-boot-starter-cache dependency has been removed from POM.xml and application.properties files. I went some more depth and removed some related annotations.

binitshrest committed:
Merge remote-tracking branch 'origin/dev' into dev

binitshrest committed:
Removed cacheable and cacheconfig of spring-boot-starter-cache

binitshrest committed:
Merge branch 'dev' into dev

ruanwenjun commented on dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/BaseDataSourceParamDTO.java:
```diff
@@ -117,7 +117,7 @@ public void setHostAndPortByAddress(String address) {
 
         this.host = String.join(Constants.COMMA, hosts);
-        this.port = Integer.parseInt(portString);
+        this.port = (Integer) Integer.parseInt(portString);
```
This change is not needed

ruanwenjun commented on dolphinscheduler-standalone-server/src/main/resources/application.yaml:
```diff
@@ -1,4 +1,4 @@
-#
+  #
```
This change is not changed.

binitshrest committed:
Removed cacheable and cacheconfig of removed spring-boot-starter-cache from ds-master

binitshrest committed:
Merge remote-tracking branch 'origin/dev' into dev

binitshrest committed:
Removed cacheable and cacheconfig of removed spring-boot-starter-cache dependencies

binitshrest commented:
@caishunfeng Can you please guide me to fix 5 failing issues?

SbloodyS commented:
You should check these two dependency. @binitshrest 
```
log4j-api-2.17.2.jar
log4j-to-slf4j-2.17.2.jar
```

binitshrest commented:
> You should check these two dependency. @binitshrest
> 
> ```
> log4j-api-2.17.2.jar
> log4j-to-slf4j-2.17.2.jar
> ```



> log4j-to-slf4j-2.17.2.jar

Thank you SbloodyS. I found them.  There are so many of these dependencies. Should i remove them too.

binitshrest commented:
> > You should check these two dependency. @binitshrest
> > ```
> > log4j-api-2.17.2.jar
> > log4j-to-slf4j-2.17.2.jar
> > ```
> 
> > log4j-to-slf4j-2.17.2.jar
> 
> Thank you @SbloodyS. I found them. There are so many of these dependencies. Should i remove them too.

binitshrest committed:
Removed cacheable and cacheconfig of removed spring-boot-starter-cache dependencies

binitshrest committed:
Removed cacheable and cacheconfig of removed spring-boot-starter-cache dependencies

ruanwenjun commented on dolphinscheduler-api/pom.xml:
```diff
@@ -121,10 +121,6 @@
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-starter-tomcat</artifactId>
                 </exclusion>
-                <exclusion>
-                    <groupId>org.apache.logging.log4j</groupId>
-                    <artifactId>log4j-to-slf4j</artifactId>
-                </exclusion>
```
Why remove this?

ruanwenjun commented on dolphinscheduler-bom/pom.xml:
```diff
@@ -927,10 +927,6 @@
                         <groupId>org.apache.logging.log4j</groupId>
                         <artifactId>log4j-core</artifactId>
                     </exclusion>
-                    <exclusion>
-                        <groupId>org.apache.logging.log4j</groupId>
-                        <artifactId>log4j-api</artifactId>
-                    </exclusion>
```
Why remove this?

binitshrest commented:
Hi @ruanwenjun,
Sure. I will revert this change asap and will work on this bug. I will appreciate a help from you? My project is failing to build with docker setup and other env setup.

binitshrest committed:
Revert "Reverting some codes that is bugging"
This reverts commit cc81bbb257b5e3c0965ca954c1d2aa14855837a3.

binitshrest commented:
![image](https://github.com/user-attachments/assets/443fcbeb-79f9-46cf-82e4-e5b8f936c40d)
Hi @ruanwenjun , the build is succesfully complete.

binitshrest committed:
Revert "Reverting some codes that is bugging"
This reverts commit cc81bbb257b5e3c0965ca954c1d2aa14855837a3.

SbloodyS committed:
Merge branch 'dev' into dev

binitshrest committed:
Merge branch 'dev' into dev

binitshrest commented:
@SbloodyS Because of so many vulnerabilities found in this project i.e due to old dependencies of mavens. The codebase is failing to build.  please allow me to change all the dependencies or any suggestion will appreciated.

ruanwenjun commented on dolphinscheduler-ui/pnpm-lock.yaml:
```diff
@@ -73,7 +57,7 @@ dependencies:
   js-cookie: 3.0.5
   lodash: 4.17.21
-  monaco-editor: 0.34.1
+  monaco-editor: 0.50.0
```
Please revert this file, this change is not related with this PR.

binitshrest replied:
sure.

SbloodyS committed:
Merge branch 'dev' into dev

binitshrest committed:
renewed monaco-editor version

binitshrest committed:
Merge remote-tracking branch 'origin/dev' into dev

binitshrest committed:
added some changes that was removed earlier

binitshrest committed:
reviewed some changes

binitshrest committed:
reviewed some changes and updated it

SbloodyS commented on dolphinscheduler-ui/pnpm-lock.yaml:
```diff
@@ -14,7 +14,7 @@
 # specific language governing permissions and limitations
 # under the License.
-lockfileVersion: 5.4
+lockfileVersion: 5.3
```
Please do not change this.

binitshrest replied:
Sure. On it.

binitshrest committed:
changes made that was asked

binitshrest committed:
read the documentatation of tools->release and implemented

binitshrest committed:
kept some files in gitignore

binitshrest committed:
changing readme file

ruanwenjun commented on .gitignore:
```diff
@@ -55,3 +55,6 @@ dolphinscheduler-master/logs
 dolphinscheduler-api/logs
 __pycache__
 ds_schema_check_test
+.env.txt
+venv
+myenv
```
Please remove this

binitshrest replied:
Sure. Sir. Thank you. I will also try to find a way to exclude these log4j-api and log4j-to-slf4j. I was looking on the same too.

ruanwenjun replied:
We need to find out where these two dependencies are imported, and exclude these, otherwise the ci will failed.

binitshrest replied:
Sure. Sir.

ruanwenjun reviewed (commented):
CI failed, since we have imported `log4j-api` and `log4j-to-slf4j` I didn't find out how to exclude these.

binitshrest committed:
made changes

binitshrest committed:
made changes

binitshrest committed:
removed extra lines

binitshrest committed:
removed log4j-api mvn repo from dol-dist>release-docs>licence

binitshrest committed:
Removed log4j-api and log4j-to-slf4j from microbench pom.xml i.e transitive dependency

SbloodyS committed:
Merge branch 'dev' into dev

binitshrest committed:
exlcuded all the transitive dependency that has log4j-api and log4j-to-slf4j

binitshrest committed:
changes are added

binitshrest committed:
add some changes

ruanwenjun commented on dolphinscheduler-storage-plugin/dolphinscheduler-storage-s3/pom.xml:
```diff
@@ -50,6 +50,16 @@
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is provided.

ruanwenjun commented on dolphinscheduler-storage-plugin/dolphinscheduler-storage-oss/pom.xml:
```diff
@@ -43,6 +43,16 @@
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is provided.

ruanwenjun commented on dolphinscheduler-storage-plugin/dolphinscheduler-storage-obs/pom.xml:
```diff
@@ -43,6 +43,16 @@
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is provided.

ruanwenjun commented on dolphinscheduler-storage-plugin/dolphinscheduler-storage-hdfs/pom.xml:
```diff
@@ -50,6 +50,16 @@
             <artifactId>dolphinscheduler-common</artifactId>
             <version>${project.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is provided.

ruanwenjun commented on dolphinscheduler-storage-plugin/dolphinscheduler-storage-gcs/pom.xml:
```diff
@@ -43,6 +43,16 @@
             <artifactId>dolphinscheduler-common</artifactId>
             <version>${project.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is provided.

ruanwenjun commented on dolphinscheduler-storage-plugin/dolphinscheduler-storage-cos/pom.xml:
```diff
@@ -36,6 +36,12 @@
             <artifactId>dolphinscheduler-common</artifactId>
             <version>${project.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is provided.

ruanwenjun commented on dolphinscheduler-storage-plugin/dolphinscheduler-storage-abs/pom.xml:
```diff
@@ -43,6 +43,12 @@
             <artifactId>dolphinscheduler-common</artifactId>
             <version>${project.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is provided.

ruanwenjun commented on dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-zookeeper/pom.xml:
```diff
@@ -96,6 +96,16 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is test.

ruanwenjun commented on dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-jdbc/pom.xml:
```diff
@@ -97,6 +97,16 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is test.

ruanwenjun commented on dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-it/pom.xml:
```diff
@@ -35,6 +35,16 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is test.

ruanwenjun commented on dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/pom.xml:
```diff
@@ -74,6 +74,16 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is test.

ruanwenjun commented on dolphinscheduler-master/pom.xml:
```diff
@@ -283,6 +268,12 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since this scope is test.

ruanwenjun commented on dolphinscheduler-extract/dolphinscheduler-extract-worker/pom.xml:
```diff
@@ -34,10 +34,30 @@
         <dependency>
             <groupId>org.apache.dolphinscheduler</groupId>
             <artifactId>dolphinscheduler-extract-base</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Do we need to add exclusions here? since these has been managed by dependencyManagement under root pom.xml.

binitshrest commented:
@ruanwenjun sir. Did i solve this issue? Please let me know. I am encouraged that all the checks are passed.

binitshrest committed:
optimize codes

binitshrest committed:
changes made

binitshrest committed:
made some changes

ruanwenjun commented:
> @ruanwenjun sir. Did i solve this issue? Please let me know. I am encouraged that all the checks are passed.

@binitshrest Yes, the ci has passed. But there are some extra things need to do.
1. Please remove two license files 
```
`dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-api-2.11.2.txt`
`dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-core-2.11.2.txt`
```
Since you have removed these from LICENSE, and we really don't rely on these, good job, you have solved a new problem.
2. I am not sure if we need to add exclusion in all place, I add comment. Since once we add these, it will be hard to remove in the future, my experience is we only need to exclude these in dependencymanagement. It's need to have a try.

binitshrest committed:
made some changes

binitshrest commented:
> > @ruanwenjun sir. Did i solve this issue? Please let me know. I am encouraged that all the checks are passed.
> 
> @binitshrest Yes, the ci has passed. But there are some extra things need to do.
> 
> 1. Please remove two license files
> 
> ```
> `dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-api-2.11.2.txt`
> `dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-core-2.11.2.txt`
> ```
> 
> Since you have removed these from LICENSE, and we really don't rely on these, good job, you have solved a new problem. 2. I am not sure if we need to add exclusion in all place, I add comment. Since once we add these, it will be hard to remove in the future, my experience is we only need to exclude these in dependencymanagement. It's need to have a try.

Yes, Sir. We don't need to add that exclusion code in all place. I saw in this morning. Once i put that exclusion in root dependency. It will remove from all. I am not sure if i should remove from test scope . I will make also some changes what you have said earlier.

binitshrest committed:
made some changes

binitshrest committed:
removed duplicate codes

binitshrest committed:
Update pom.xml

binitshrest committed:
removed two licence file log4j-api & log4j-to-slf4j as asked

ruanwenjun committed:
Merge branch 'dev' into dev

binitshrest commented:
> Please retry analysis of this Pull-Request directly on SonarCloud

Do i have to worry about this sir? @ruanwenjun

ruanwenjun commented:
> > Please retry analysis of this Pull-Request directly on SonarCloud
> 
> Do i have to worry about this sir? @ruanwenjun

No, this is not related to your change, and will not block this PR.

SbloodyS commented on .gitignore:
```diff
@@ -54,4 +54,4 @@ dolphinscheduler-worker/logs
 dolphinscheduler-api/logs
 __pycache__
-ds_schema_check_test
+ds_schema_check_test
```
Please avoid unnessnary chanage.

SbloodyS replied:
This is still unaddressed.

SbloodyS reviewed (commented):
Generally LGTM. Just some NIT.

binitshrest commented:
This issue was closed accidentally. I am reopening this issue.

binitshrest committed:
made some changes

binitshrest committed:
Merge remote-tracking branch 'origin/dev' into dev

binitshrest committed:
made some changes

binitshrest commented:
> This issue was closed accidentally. I am reopening this issue.

Reopening

binitshrest committed:
made some changes

SbloodyS reviewed (approved):
+1

ruanwenjun reviewed (approved):
+1
